### PR TITLE
feat(icon-button-bar): add divider prop with demo and tests

### LINKED ISSUE #568: Divider between buttons in IconButtonBar
### Details

- **Application / team -** Advisor

### Proposal

The first two buttons in our IconButtonBar is used to switch between views. The third button can be used on either view, so we've added a "divider" before the third button. This divider is actually just a left border on the third button. It would be nice to have an actual divider element between the buttons because on focus the buttons' border covers the "divider".

image

### Goals / use cases

- [ ] Be able to add a divider between buttons

### PR DESCRIPTION
**Pull request by jendowns**
## Affected issues

- Resolves #568 

## Proposed changes

- add `divider` prop that renders a divider element on either side of an icon button, depending on prop value
- add storybook knobs/demo for new prop
- add tests for new prop

## Testing instructions

-

### CODE REVIEW

File: src/components/IconButtonBar/IconButtonBar.js [hunk 1]
```diff
@@ -41,14 +41,25 @@ const IconButtonBar = ({
...
-      key={action.label || `${namespace}__button--icon`}
-      size={size}
-      tooltip={tooltip}
-      tooltipDirection={iconTooltipDirection}
-    />
+  const renderIconButton = (action, index) => (
+    <Fragment key={action.label || `${namespace}__button--icon--${index}`}>
+      {(action.divider === 'left' || action.divider === 'sides') && (
+        <span className={`${namespace}__divider`} aria-hidden>
+          <span className={`${namespace}__divider__inner`} aria-hidden />
```
Comment (SimonFinney) [inline src/components/IconButtonBar/IconButtonBar.js:48]: Awesome! As I understand it, the `aria-hidden` declaration on the parent `divider` element should be sufficient to remove this from the accessibility tree.

... Unless different screen readers handle this in different ways? What's the rationale for adding it to the `inner` element?

File: src/components/IconButtonBar/IconButtonBar.stories.js [hunk 2]
```diff
@@ -35,11 +35,16 @@ import { TooltipDirection } from '../IconButton/IconButton';
 storiesOf(patterns('IconButtonBar'), module).add(
   'default',
   () => {
-    const size = select('size', sizes, sizes[0]);
+    const size = select('size', sizes, 'sm');
     /* eslint-disable no-nested-ternary */
     const actions = [
       {
         className,
+        divider: select(
```
Comment (SimonFinney) [inline src/components/IconButtonBar/IconButtonBar.stories.js:43]: Would you mind including a `null` / `undefined` option to provide an example of the `IconButtonBar` without any dividers?

File: src/components/IconButtonBar/_index.scss [hunk 3]
```diff
@@ -49,4 +49,16 @@
       width: $icon-button-bar__sizing__dimensions;
     }
   }
+
+  &__divider {
+    background-color: transparent;
```
Comment (SimonFinney) [inline src/components/IconButtonBar/_index.scss:54]: Perhaps I'm overlooking something, but at first glance, this appears to be redundant - are we able to safely remove this or does it cover something specific?

Comment (jendowns) [reply src/components/IconButtonBar/_index.scss:54]: This style rule is for clarity and comprehension.

**Comment by jendowns** [comment]
Thanks @SimonFinney this is ready to review again 👍

File: src/components/IconButtonBar/_index.scss (hunk 3 above)
Comment (SimonFinney) [reply src/components/IconButtonBar/_index.scss:54]: If it's for clarity for subsequent developers, would it be more suitable to indicate this via a comment? That way it wouldn't be included in the build output.

**Pull request merged**